app/models/concerns/date_helpers.rb

```diff
@@ -0,0 +1,13 @@
+module DateHelpers
+
+  private
+
+  def valid_date?(date)
+    Date.parse(date.to_s)
+  rescue ArgumentError
+    return false
+  else
```

Comment:
This seems redundant and a bit awkward. First, `Date.parse` will return a truthy value if the string encodes a valid date, so the explicit return of `true` is not necessary. Second, the logic flow is "do something -> if it failed, send failure message -> if it didn't, send success message". More natural, for me at least, would be "do something and send success message -> it that failed, send failure message".